source/extensions/clusters/aggregate/cluster.h

```diff
@@ -0,0 +1,156 @@
+#pragma once
+
+#include "envoy/config/cluster/aggregate/cluster.pb.h"
+#include "envoy/config/cluster/aggregate/cluster.pb.validate.h"
+
+#include "common/upstream/cluster_factory_impl.h"
+#include "common/upstream/upstream_impl.h"
+
+#include "extensions/clusters/aggregate/lb_context.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+
+using PriorityContext = std::pair<Upstream::PrioritySetImpl,
+                                  std::vector<std::pair<uint32_t, Upstream::ThreadLocalCluster*>>>;
+
+class Cluster : public Upstream::BaseDynamicClusterImpl, Upstream::ClusterUpdateCallbacks {
```

Comment:
i dont think you need `BaseDynamicClusterImpl`, `ClusterImplBase` might be good enough?